DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,121 @@
+| Review Count:   | 0                                                               |
+| Author:         | Jack Stouffer <jack@jackstouffer.com>                           |
+| Implementation: | N/A                                                             |
+| Status:         |                                                                 |
+
+## Abstract
+
+In order to incrementally improve D or it's standard library, it's often necessary to
+mark features or functions for future removal. This document proposes a standardized
+process for language maintainers to remove public features. This process would be 
+used across DMD, Druntime, and Phobos.
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+* [Acknowledgements](#acknowledgements)
+* [Reviews](#reviews)
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends up being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, so as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Users must be given at least 10 non-patch releases before the deprecated features
+are removed. More releases can be given if the removed code is commonly used.
+There are two cases where the deprecation period is allowed to be shorter:
+
+1. The code or feature is notably dangerous or unsafe, and users need to remove
+it from their code as soon as possible.
+2. The existance of the current code precludes its own fix or the fix of an equally
+important issue.
+
+Shorting the deprecation period should be done with caution to avoid giving D
+an image of instability.
+
+At the time of the pull request for deprecation, all code in Phobos, Druntime,
+and DMD must be updated to remove use of the effected code. Any projects that
+are tested on the Project Tester that are broken should also have their
+maintainers notified.
+
+Both at the time of deprecation and removal, a changelog entry must be made. This
+changelog entry should have a short motivation for the deprecation (or removal)
+and should to describe what steps can be taken by the user to upgrade their codebase.
+
+In order to facilitate on schedule deprecations, a comment of the format
+`@@@DEPRECATED_[version]@@@` should be added to the top of the code to be removed/disabled.
+This comment allows code to be easily searched before every release to
+catch all planned deprecations.
+
+### Public Functions, Types, and Modules
+
+All removals (or changes that make the code `private`/`package`) of public functions,
```

Comment:
"All removals (or changes that make the code `private`/`package`) of public functions, types, and modules must be accompanied with a deprecation period." -> When public functions, types, and modules are removed, or their protection attributes are made more restrictive, the changes must be accompanied by a deprecation period.

I notice this leaves out variables. Would it be correct to say "public symbols" rather than "functions, types and modules"?